src/Models/BaseModel.js

```diff
@@ -1,96 +1,65 @@
 /**
- * Abstract Model Class BaseModel
+ * File Auto-Generated by the RICADO Gen 4 PHP API Project
+ * 
+ * Do Not Edit this File Manually!
+ */
+
+/**
+ * The Abstract **BaseModel** Class
  * @abstract
  * @hideconstructor
  */
-class BaseModel {
-  /**
+class BaseModel
+{
+    /**
      * BaseModel Constructor
-     *
-     * @package
-     * @param {object} args - The Model Arguments
-     */
-  constructor({ json })
-  {
-    this._json = json;
-    this._updateJson = {};
-  }
-
-  /**
-     * Properties
+     * 
+     * @public
+     * @param {Object} json - The Model Properties
      */
+    constructor(json)
+    {
+        /**
+         * The Model Properties
+         * 
+         * @type {Object<string, any>}
+         * @protected
+         */
+        this._json = json;
```

Comment:
@sadsa As discussed, this will likely be to do with the `T` type being used in WritableDraft<T> getting a typeof CompanyModel rather than the actual Class itself?

I'll mark as resolved and leave this with you to investigate in the library / application you are using.